src/v/storage/BUILD

```diff
@@ -165,11 +165,32 @@ redpanda_cc_library(
 
 redpanda_cc_library(
     name = "chunk_cache",
+    srcs = [
+        "chunk_cache.cc",
+    ],
     hdrs = [
         "chunk_cache.h",
         "segment_appender_chunk.h",
     ],
+    implementation_deps = [
+        "//src/v/config",
+        "//src/v/resource_mgmt:memory_groups",
+        "@boost//:iterator",
+    ],
     include_prefix = "storage",
+    visibility = [
+        "//src/v/storage/tests:__pkg__",
+        # TODO(bazel) chunk cache could probably be instantiated in
+        # log_manager.cc instead of in redpanda::application.
+        "//src/v/redpanda:__pkg__",
```

Comment:
Just curious, what's the significance of these? Is the TODO that we eventually want to remove redpanda:__pkg__ from this list and instead just make it a dependency of v/storage?

Reply:
yeah limiting the visibility means that this is an implementation detail of the storage layer instead of public API